api/v1alpha1/wasm_types.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// Wasm defines a wasm extension.
+type Wasm struct {
+	// Name is a unique name for this Wasm extension. It is used to identify the
+	// Wasm extension if multiple extensions are handled by the same vm_id and root_id.
+	// It's also used for logging/debugging.
+	Name string `json:"name"`
+
+	// VmID is an ID that will be used along with a hash of the wasm code to
+	// determine which VM will be used to load the Wasm extension. All extensions
+	// that have the same vm_id and code will use the same VM.
+	//
+	// Note that sharing a VM between plugins can reduce memory utilization and
+	// make sharing of data easier, but it may have security implications.
+	VmID *string `json:"vmID,omitempty"`
+
+	// RootID is a unique ID for a set of extensions in a VM which will share a
+	// RootContext and Contexts if applicable (e.g., an Wasm HttpFilter and an Wasm AccessLog).
+	// If left blank, all extensions with a blank root_id with the same vm_id will share Context(s).
+	RootID *string `json:"rootID,omitempty"`
+
+	// Code is the wasm code for the extension.
+	Code WasmCodeSource `json:"code"`
+
+	// Configuration for the wasm code.
+	Config *WasmConfig `json:"config"`
+
+	// InsertBefore is the name of the filter that this Wasm extension should be
+	// inserted before.
+	// If not specified, this Wasm extension will be inserted before the router filter.
+	InsertBefore *EnvoyFilter `json:"insertBeforeFilter"`
```

Comment:
This depends om https://github.com/envoyproxy/gateway/pull/2993

Reply:
here is my suggestion for prioritization of dealing with filter ordering to so we move faster in the API reviews
cc @guydc 
1. lets find a default home (order) for `ext_proc` and `wasm` (maybe right after `ext_auth`)
2. lets create per proxy ordering (which you have already done with #2993 (love it)) once we get use cases in
3. lets then consider creating per wasm location override once we get use cases in

Reply:
@arkodg We may need to address ordering first because different WASM extensions should be in different locations in the chain.

Reply:
yeah we can tackle before v1.1 , but it doesnt need to block this PR